File: Algorithm/linear_search/linear_search.py
def linearSearch(list, target):
    result = None
    #returns the index if found, otherwise return None
    for i in range(len(list)):
        if list[i] == target:
            result = i
            break

    return result

def verify(result):
    #we use is keyword to check an object's identity, since None is not technically a value (there is no value in variable result in the first place)
    if result is not None: 
        print("Found at index", result)
    else:
        print("Not Found")

#main
number = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print("list =", number)

target = int(input("target = "))

verify(linearSearch(number, target))

File: Algorithm/linear_search/linear_search.cpp
#include <bits/stdc++.h>
using namespace std;

int linearSearch(vector<int> v, int target) {
    int result = -1;
    for (int i = 0; i < v.size(); i++) {
        if (v[i] == target) {
            result = i;
        }
    }
    return result;
}

void verify(int result) {
    if (result != -1) {
        cout << "Found at index " << result << endl;
    } else {
        cout << "Not Found" << endl;
    }
}

int main() {
    vector<int> v = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    cout << "vector = ";
    for (int i = 0; i < v.size(); i++) {
        cout << v[i] << (i < v.size() - 1 ? ", " : "");
    }

    cout << endl;
    int target;
    cout << "target = "; cin >> target;
    verify(linearSearch(v, target));
    
    return 0;
}